### PR TITLE
#157612278 Fetch all requests endpoint

### PR DESCRIPTION
**Pull request by omobosteven**
#### What does this PR do?
- Create API endpoint to fetch all the requests of a logged in user 

#### Description of Task to be completed?
- User should be able to fetch all his/her request through this endpoint

#### How should this be manually tested?
- Clone the project locally from git clone https://github.com/omobosteven/maintenance-tracker.git
- Navigate to the project in your terminal
- Install all the necessary dependencies as stated in the README file
- run $ npm install
- Run app locally using $ npm run start-dev
- Open postman and enter the URL http://localhost:3000/api/v1/requests

#### Any background context you want to provide?
- none

#### What are the relevant pivotal tracker stories?
- story type: features
- story id: 157612278
- story title: User should be able to get all his/her request

#### ScreenShots:
- none

#### What are the relevant Github Issues?
- none

#### Questions:
- none

### CODE REVIEW

File: server/controllers/RequestsController.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   * @description Fetch all the requests of a logged in user
+   *
+   * @param {Object} req - HTTP Request
+   * @param {Object} res - HTTP Response
+   *
+   * @return {Object} Returned object
+   */
+  static getRequests(req, res) {
+    const userId = 1;
+    const myRequests = requests.filter(request => request.userId === userId);
```
Comment (nebanat) [inline server/controllers/RequestsController.js]: What happens when there are no requests for that user? what do you return?

File: server/routes/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import requests from './requests';
+import notFound from './notfound';
+
+const router = express.Router();
+
+router.use('/api/v1', requests);
+router.use('/', notFound);
```
Comment (nebanat) [inline server/routes/index.js]: why does the home page point to notFound?

File: server/tests/notfound.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { describe, it } from 'mocha';
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../app';
+
+chai.use(chaiHttp);
+
+describe('Test for unhandled endpoint', () => {
+  it('should return a message if endpoint is not handled', (done) => {
```
Comment (nebanat) [inline server/tests/notfound.js]: This test is ambiguous, what do you mean unhandled endpoint?

File: server/controllers/RequestsController.js (hunk 1 above)
Comment (omobosteven) [reply server/controllers/RequestsController.js]: I'll fix that

File: server/routes/index.js (hunk 2 above)
Comment (omobosteven) [reply server/routes/index.js]: The home page does not point to not found, I took care of the home page in the app.js file.
my home is http://localhost:3000/, I'm trying catch route that does not exist.

File: server/tests/notfound.js (hunk 3 above)
Comment (omobosteven) [reply server/tests/notfound.js]: I'll fix that

File: server/controllers/RequestsController.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  /**
+   * @description Fetch all the requests of a logged in user
+   *
+   * @param {Object} req - HTTP Request
+   * @param {Object} res - HTTP Response
+   *
+   * @return {Object} Returned object
+   */
+  static getRequests(req, res) {
+    const userId = parseInt(req.headers['user-id'], 10);
```
Comment (nebanat) [inline server/controllers/RequestsController.js]: Try to handle more errors and edge cases. what happens when the user-id is a string?

File: server/controllers/RequestsController.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+   * @param {Object} req - HTTP Request
+   * @param {Object} res - HTTP Response
+   *
+   * @return {Object} Returned object
+   */
+  static getRequests(req, res) {
+    /**
+     * handles error if requests data is an empty object
+     */
+    const isEmpty = Object.keys(requests);
```
Comment (omobosteven) [inline server/controllers/RequestsController.js]: Trying to check if the requests dummy data is an empty object

**Pull request merged**